### PR TITLE
Add drag-entered and drag-exited events to Tray

### LINKED ISSUE #3094: [request] Tray hover event
Creating feature request for adding a hover event for [Tray](https://github.com/atom/electron/blob/master/docs/api/tray.md). Is it something that has already been discussed (*couldn't find in the search at least*) and rejected or are there platform limitations?

**Use case**
Change (*or [animate](https://twitter.com/RenarsVilnis/status/654638564328767488)*) icon when files are going get drag-n-dropped onto the tray to display reaction instead of the current behavior of changing the appearance of the cursor.

### PR DESCRIPTION
**Pull request by nishanths**
Resolves #3094

### CODE REVIEW

**Comment by zcbenz** [comment]
Can you rename the events to `drag-enter` and `drag-leave`? So we can match the names of DOM's `dragenter` and `dragleave` events.

We should also add `drag-end` and `drop` events, which are necessary for implementing full-featured hover effect.

**Comment by nishanths** [comment]
- [ ] rename events to `drag-enter` and `drag-leave`
- [ ] add `drag-end` and `drop` events

**Comment by nishanths** [comment]
@zcbenz Thoughts on whether the `drop` event notification occurs at the correct place?

File: atom/browser/ui/tray_icon_cocoa.mm [hunk 1]
```diff
@@ -254,9 +254,23 @@ - (void)rightMouseUp:(NSEvent*)event {
...
+- (void)draggingExited:(id <NSDraggingInfo>)sender {
+  trayIcon_->NotifyDragExited();
+}
+
+- (void)draggingEnded:(id <NSDraggingInfo>)sender {
+  trayIcon_->NotifyDragEnded();
+}
+
+- (BOOL)prepareForDragOperation:(id <NSDraggingInfo>)sender {
+  trayIcon_->NotifyDrop();
```
Comment (nishanths) [inline atom/browser/ui/tray_icon_cocoa.mm]: @zcbenz Thoughts on whether the drop event notification occurs at the correct place?

**Comment by zcbenz** [comment]
Perfect, thanks!

**Pull request merged**